### PR TITLE
Fixes #1477 - Hostgroup no longer looses puppet classes when environm…

### PR DESCRIPTION
**Pull request by xprazak2**
…ent changed during hostgroup editing

### CODE REVIEW

File: app/views/puppetclasses/_class_selection.html.erb [hunk 1]
```diff
@@ -45,7 +45,7 @@
     <div class="row">
       <%= render "puppetclasses/classes",
         :puppetclasses => obj.available_puppetclasses,
-        :selected_puppet_classes => obj.classes,
+        :selected_puppet_classes => obj.puppetclasses,
```
Comment (domcleal) [inline app/views/puppetclasses/_class_selection.html.erb]: .classes is important as it includes any classes inherited from a host group (in the case of host#edit) or a parent (in hostgroup#edit)

**Comment by xprazak2** [comment]
I did not realize puppetclasses can be inherited from ancestors, this should fix it.

**Comment by dLobatog** [comment]
This is missing some test to make sure there are no regressions, could you add one @xprazak2 ?

**Comment by xprazak2** [comment]
I've added the test.

**Comment by dLobatog** [comment]
Merged as 521ef9c4025be617beafd0c978c9ba7ed98fa95c, thanks @xprazak2!

**Pull request closed without merging**